src/libse/Common/FormattableTextReader.cs

```diff
@@ -0,0 +1,82 @@
+namespace Nikse.SubtitleEdit.Core.Common
+{
+    public struct FormattableTextReader
```

Comment:
I already see several use case for this `FormattableTextReader`

Reply:
The struct is also used instead of class to avoid overheat that can be caused by class.

For example in method `IsAllUppercase`, let say is being called inside a loop to check if text in 
paragraphs are all uppercase then no allocation will happen due to struct being store in stack instead of heap!